Controller/Subscriber/Subscribe.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Smaily\SmailyForMagento\Controller\Subscriber;
+
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\RequestInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Smaily\SmailyForMagento\Helper\Data as Helper;
+
+/**
+ * Sends new subscribers from newsletter sign-up form directly to Smaily.
+ */
+class Subscribe
+{
+    /**
+     * Smaily helper class contains all methods.
+     *
+     * @var Smaily\SmailyForMagento\Helper\Data
+     */
+    protected $helper;
+
+    /**
+     * Gathers extra data from store.
+     *
+     * @var Magento\Store\Model\StoreManagerInterface
+     */
+    private $_storeManager;
+
+    /**
+     * Gathers extra data if customer is logged in when subscribing.
+     *
+     * @var Magento\Customer\Model\Session
+     */
+    private $_customerSession;
+
+    /**
+     * Gather post parameters (Name field from form).
+     *
+     * @var Magento\Framework\App\RequestInterface
+     */
+    private $_request;
+
+    public function __construct(
+        Helper $helper,
+        RequestInterface $request,
+        StoreManagerInterface $storeManager,
+        Session $customerSession
+    ) {
+        $this->helper = $helper;
+        $this->_request = $request;
+        $this->_storeManager = $storeManager;
+        $this->_customerSession = $customerSession;
+    }
+
+    /**
+     * Send subscriber data to Smaily.
+     *
+     * @param \Magento\Newsletter\Model\Subscriber $subject
+     * @param string $email
+     * @return null
+     */
+    public function beforeSubscribe(\Magento\Newsletter\Model\Subscriber $subject, $email)
+    {
+        // Check Smaily extension/newsletter subscribers collection are enabled.
+        if ($this->helper->isEnabled() && $this->helper->isNewsletterSubscriptionEnabled()) {
+            $autoresponderId = $this->helper->getGeneralConfig('autoresponder_id');
+            $name = '';
+            if ($this->_request->getPost('name')) {
+                $name = (string) $this->_request->getPost('name');
+            }
+            // Create addtional fields array.
+            $extra = [
+                'name' => $name,
+                'subscription_type' => 'Subscriber',
```

Comment:
As we discussed - let's remove `subscription_type`, and add `customer_id` with default value of empty string to the list of additional fields.

Reply:
Removed `subscription_type` and added `customer_id` with initial value as empty string. 1f70722238b87dbad1f0c8d232f122d37f41e006